guillotina/contentnegotiation.py

```diff
@@ -1,844 +1,34 @@
-                # wildcards, so content types like */* will match appropriately.  We ge
-                # back from this the concrete AcceptParameter as specified by the server
-                # if there is a match, so we know the result contains no unintentional wildcards
-                match = self._contains_match(p, server)
-                if match is not None:
-                    # if there is a match, register it
-                    allowable.append(match)
-
-            log.info("Allowable: " + str(q) + ":" + str(allowable))
-
-            # we now know if there are 0, 1 or many allowable content types at this q value
-            if len(allowable) == 0:
-                # we didn't find anything, so keep looking at the next q value
-                continue
-            elif len(allowable) == 1:
-                # we found exactly one match, so this is our content type to use
-                return allowable[0]
-            else:
-                # we found multiple supported content types at this q value, so now we need
-                # to choose the server's preference
-                for i in range(len(server)):
-                    # iterate through the server explicitly by numerical position
-                    if server[i] in allowable:
-                        # when we find our first content type in the allowable list, it is
-                        # the highest ranked server content type that is allowable, so
-                        # this is our type
-                        return server[i]
-
-        # we've got to here without returning anything, which means that the client and
-        # server can't come to an agreement on what content type they want and can deliver.
-        # There's nothing more we can do!
-        return None
-
-
-@implementer(IContentNegotiation)
-class ContentNegotiatorUtility(object):
-
-    def __init__(self, header, enabled_values):
-        if header == 'content_type':
-            server = [AcceptParameters(content_type=ContentType(x)) for x in enabled_values]
-            self.cn = ContentNegotiator(acceptable=server)
-        if header == 'language':
-            server = [AcceptParameters(language=Language(x)) for x in enabled_values]
-            self.cn = ContentNegotiator(acceptable=server)
-
-    def negotiate(self, **kwargs):
-        return self.cn.negotiate(**kwargs)
-
-
-def content_type_negotiation(request, resource, view):
+def get_ordered_content_type_value_list(value):
+    '''
+    ca-ES,ca;q=0.9,en;q=0.8,es;q=0.7
+    '''
+    result = []
+    for ct in value.lower().split(','):
+        weight = 1.0
+        ct_name = ct.split(';')[0]
+        for param in ct.split(';')[1:]:
+            if param.startswith('q='):
+                try:
+                    weight = float(param[len('q='):])
+                except ValueError:
+                    pass
+        result.append((weight, ct_name))
+    return [i[1] for i in sorted(result, reverse=True)]
```

Comment:
maybe we could use the _sort_by_q deleted function so we get the real order?

Reply:
I'm not familiar with what you're talking about here. Can you clarify?

Reply:
If u check the function from the old code u will see the logic on the priority to define which is prior to which one

Reply:
ah yes, I see. I'll fix

Reply:
the old code is so horrible